[P151] Add Honeywell I2C Pressure sensor plugin

TD-er opened the pull request:
[I2C protocol description](https://prod-edam.honeywell.com/content/dam/honeywell-edam/sps/siot/en-us/products/sensors/pressure-sensors/board-mount-pressure-sensors/common/documents/sps-siot-i2c-comms-digital-output-pressure-sensors-tn-008201-3-en-ciid-45841.pdf?download=false)
[Sensor description on Honeywell site](https://sps.honeywell.com/us/en/products/advanced-sensing-technologies/healthcare-sensing/board-mount-pressure-sensors/trustability-hsc-series)
[Datasheet](https://prod-edam.honeywell.com/content/dam/honeywell-edam/sps/siot/en-us/products/sensors/pressure-sensors/board-mount-pressure-sensors/trustability-hsc-series/documents/sps-siot-trustability-hsc-series-high-accuracy-board-mount-pressure-sensors-50099148-a-en-ciid-151133.pdf?download=false)

Pin numbering per package starting at page 19

![image](https://user-images.githubusercontent.com/3751318/230880185-a4c59f30-b3a1-420e-948a-f29e4ff9a775.png)


ToDo:
- [x] Test with real hardware
- [ ] Documentation
- [ ] (optional) Add decoder for elaborate Honeywell part numbering schema.

TD-er committed:
[P151] Add Honeywell I2C Pressure sensor plugin

tonhuisman commented on src/P151_Honeywell_pressure.ino:
```diff
@@ -0,0 +1,162 @@
+        0x88 >> 1,
+        0x98 >> 1
+      };
+      constexpr size_t addrLength = sizeof(i2cAddressValues) / sizeof(uint8_t);
```
I think a distinction between `PLUGIN_I2C_HAS_ADDRESS` and `PLUGIN_WEBFORM_SHOW_I2C_PARAMS` has to be added here to make the device appear in the I2C Scanner.

TD-er replied:
Check!
Will look into it.
Right now the state of this PR is: it compiled :)

TD-er committed:
[P151] Fix I2C scanner result not showing address selection dialog

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
[Honeywell] Fix measuring temperature and pressure on real sensor

tonhuisman commented on src/P151_Honeywell_pressure.ino:
```diff
@@ -0,0 +1,168 @@
+      // 88 (58 hex)
+      // 104 (68 hex)
+      // 120 (78 hex)
+      // 136 (88 hex)
+      // 152 (98 hex)
```
Aren't these addresses going to be 'flattened' to `0x08` and `0x18`, as these I2C addresses can only be 7 bit wide...? And also they can't be properly stored in an `int8_t` that is often used to store the _used_ address, so -1 can be used to indicate None/Unused?

TD-er replied:
The I2C helper functions like `addFormSelectorI2C` and `intArrayContains` accept `uin8_t`.

These adresses are how they also show up in the I2C scan and are noted in the datasheet and product nomenclature.

TD-er committed:
[Honeywell] Fix accepting "stale" data when value did change

Apparently the sensor also reports data to be 'stale' when it is busy performing a new measurement

TD-er committed:
Merge branch 'mega' into feature/P151_Honeywel_I2C_pressure

TD-er committed:
[Honeywell] Add documentation for P151 I2C Honeywell Pressure
